File: starters/java/src/main/java/io/fproto/core/SessionEvent.java
package io.fproto.core;

public enum SessionEvent {
    HANDSHAKE_OK("handshake_ok"),
    HANDSHAKE_FAIL("handshake_fail"),
    PACKET_RECEIVED("packet_received"),
    IDLE_TIMEOUT("idle_timeout"),
    EXPIRY_TIMEOUT("expiry_timeout"),
    EXPLICIT_LOGOUT("explicit_logout");

    private final String value;

    SessionEvent(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: starters/php/src/Session/SessionEvent.php
<?php

declare(strict_types=1);

namespace FProto\Session;

enum SessionEvent: string
{
    case HandshakeOK = 'handshake_ok';
    case HandshakeFail = 'handshake_fail';
    case PacketReceived = 'packet_received';
    case IdleTimeout = 'idle_timeout';
    case ExpiryTimeout = 'expiry_timeout';
    case ExplicitLogout = 'explicit_logout';
}
